frontend/src/components/CalcSpectrum.tsx

```diff
@@ -242,7 +262,11 @@ export const CalcSpectrum: React.FC = () => {
     />
   );
   return (
-    <form onSubmit={handleSubmit(onSubmit)}>
+    <form
+      onSubmit={handleSubmit((data) => {
+        onSubmit(data, `calculate-spectrum`);
+      })}
```

Comment:
```suggestion
      onSubmit={handleSubmit((data) => onSubmit(data, `calculate-spectrum`))}
```
This should work as well, but let me know if it leads to a type error.